docs/explanation/lifecycle.rst

```diff
@@ -2,7 +2,62 @@
 Lifecycle details
 *****************
 
+Each part is built in five separate steps:
+
+#. ``PULL`` — The components for the part are retrieved from their stated
+   location and placed into a package cache area.
+#. ``OVERLAY`` — Any overlay packages are incorporated into the staging area
+   and the overlay script is run. Finally, any overlay filters are applied.
+#. ``BUILD`` — The part is built according to the particular part plugin and
+   build override.
+#. ``STAGE`` — The specified outputs from the ``BUILD`` stage are copied into
+   a unified staging area for all parts.
+#. ``PRIME`` — The specified files are copied from the staging area to the
+   priming area for use in the final project. This is distinct from ``STAGE``
```

Comment:
The term `project` here feels a bit odd. In fact I often search for a term to use for "the set of files that are included in the actual resulting snap/rock/charm", and sometimes I use "payload" for that. Maybe we should make a term (any term, but I vote for payload as it's application-agnostic) "official"?

Reply:
Yeah I wasn't too happy with that terminology either. Payload is much better - incorporated!